cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/group/GcpInstanceGroupResourceBuilder.java

```diff
@@ -0,0 +1,141 @@
+package com.sequenceiq.cloudbreak.cloud.gcp.group;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.Compute.InstanceGroups.AddInstances;
+import com.google.api.services.compute.Compute.InstanceGroups.Delete;
+import com.google.api.services.compute.Compute.InstanceGroups.Insert;
+import com.google.api.services.compute.Compute.InstanceGroups.ListInstances;
+import com.google.api.services.compute.Compute.InstanceGroups.RemoveInstances;
+import com.google.api.services.compute.model.InstanceGroup;
+import com.google.api.services.compute.model.InstanceGroupsAddInstancesRequest;
+import com.google.api.services.compute.model.InstanceGroupsListInstances;
+import com.google.api.services.compute.model.InstanceGroupsListInstancesRequest;
+import com.google.api.services.compute.model.InstanceGroupsRemoveInstancesRequest;
+import com.google.api.services.compute.model.InstanceReference;
+import com.google.api.services.compute.model.InstanceWithNamedPorts;
+import com.google.common.collect.Sets;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.gcp.GcpResourceException;
+import com.sequenceiq.cloudbreak.cloud.gcp.context.GcpContext;
+import com.sequenceiq.cloudbreak.cloud.model.CloudInstance;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResourceStatus;
+import com.sequenceiq.cloudbreak.cloud.model.Group;
+import com.sequenceiq.cloudbreak.cloud.model.Location;
+import com.sequenceiq.cloudbreak.cloud.model.Network;
+import com.sequenceiq.cloudbreak.cloud.model.Security;
+import com.sequenceiq.common.api.type.ResourceType;
+
+public class GcpInstanceGroupResourceBuilder extends AbstractGcpGroupBuilder {
+
+    private static final int ORDER = 1;
+
+    @Override
+    public CloudResource create(GcpContext context, AuthenticatedContext auth, Group group, Network network) {
+        String resourceName = getResourceNameService().resourceName(resourceType(), context.getName());
+        return createNamedResource(resourceType(), resourceName);
+    }
+
+    @Override
+    public CloudResource build(GcpContext context,
+            AuthenticatedContext auth, Group group,
+            Network network, Security security, CloudResource resource) throws Exception {
+
+        Insert insert = context.getCompute().instanceGroups().insert(context.getProjectId(),
+                context.getLocation().getAvailabilityZone().value(), new InstanceGroup().setName(resource.getName()));
+        CloudResource instanceGroupCloudResource = doOperationalRequest(resource, insert);
+
+        if (group.getInstances().isEmpty()) {
```

Comment:
Is this something we should check before we make the GCP API call, for efficiency?

Reply:
The previous call is creating the group. We then check that there are instances before adding any, if a group can exist with instances in it then this is where we check for that.